### DISCUSSION TITLE
mapwriter: Problem with very large maps

### DISCUSSION THREAD

**Discussion by demiantres** [mapsforge/mapsforge#1226] [Q&A] [answered]
I have created a very large map (12GB). When opening the map in Cruiser I get an "Invalid size: 12002019110" error.
I suspect that the problem is within the mapsforge-writer which calculates the header size using "int" size but it should be "long".

For instance the variable "totalHeaderSize" (https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-writer/src/main/java/org/mapsforge/map/writer/MapFileWriter.java, line 361) should be "long".

I cannot get the writer plugin to compile else I would have tried to fix it by myself.

Or is the mapsforge format generally limited to a specific file size?

**Comment by devemux86** [comment, marked as answer]
_You can clone the repository locally and open it with IDEA (or Android Studio or a Gradle compatible IDE).
Then it should compile via IDE UI or even command line._

---

The offline maps are usually country sized. For large countries they are split by large sub regions.
Some of the largest countries with dense road network, like Germany or France are only 2.5 GB.

Huge map files e.g. > 4 GB are difficult to create and their real usefulness is questionable.
It is simpler to open simultaneously only the needed multiple map files.

In any case you can study the code and check what happens.
If there is some small improvement - _which does not break map compatibility_ - we could see it.

**Comment by devemux86** [comment]
By the way the writer uses `ByteBuffer` which returns parent `Buffer.position` as integer.
